src/tainting/Dataflow_tainting.ml

```diff
@@ -1172,6 +1218,37 @@ let check_tainted_return env tok e : Taints.t * Lval_env.t =
   report_findings env findings;
   (taints, var_env')
 
+let findings_from_arg_updates_at_exit enter_env exit_env : T.finding list =
+  (* TOOD: We need to get a map of `lval` to `Taint.arg`, and if an extension
+   * of `lval` has new taints, then we can compute its correspoding `Taint.arg`
+   * extension and generate an `ArgToArg` finding too. *)
+  enter_env |> Lval_env.seq_of_tainted |> List.of_seq
+  |> List.concat_map (fun (lval, enter_taints) ->
+         (* For each lval in the enter_env, we get its `T.arg`, and check
+          * if it got new taints at the exit_env. If so, we generate an
+          * ArgToArg. *)
+         match
+           enter_taints |> Taints.elements
+           |> Common.map_filter (fun taint ->
+                  match taint.T.orig with
+                  | T.Arg arg -> Some arg
+                  | _ -> None)
+         with
+         | []
+         | _ :: _ :: _ ->
+             []
+         | [ arg ] ->
+             let exit_taints =
+               Lval_env.dumb_find exit_env lval |> status_to_taints
+             in
+             let new_taints = Taints.diff exit_taints enter_taints in
+             (* TODO: Also report if taints are _cleaned_. *)
+             new_taints |> Taints.elements
+             |> Common.map_filter (fun taint ->
+                    match taint.T.orig with
+                    | T.Arg t -> Some (T.ArgToArg (t, taint.tokens, arg))
```

Comment:
Can a function have multiple facts associated with it? (e.g. a function might be an `ArgToArg` and also an `ArgToSink`)

Reply:
It can have multiple, several `ArgToArg` as well as several `ArgToSink`.